### PR TITLE
Update RTC from NTP when NTP have a bigger hour

### PR DESCRIPTION
**Pull request by Elemeants**
Signed-off-by: Polanco Avalos, Juan <juan.polanco.avalos@intel.com>

### CODE REVIEW

File: src/system/irrigation.h [hunk 1]
```diff
@@ -80,23 +80,31 @@ class SystemTimeProvider : public ITimeProvider {
...
-                if (!anySucess) {
-                    datetime = _tp->get();
+                now = _tp->get().toDateTime();
+                if (now >= prev) {
+                    datetime = now;
+                    anySucess = true;
                 }
-                anySucess = true;
             } else {
                 logger << LOG_ERROR << "Error while updating time provider!" << EndLine;
```
Comment (m-mora) [inline src/system/irrigation.h:97]: could add which time provider has an error

File: src/system/irrigation.h [hunk 2]
```diff
@@ -80,23 +80,31 @@ class SystemTimeProvider : public ITimeProvider {
 
     bool update() {
         bool anySucess = false;
+        DateTime now(getBuildTimeAsUnixTime(__DATE__, __TIME__));
+        DateTime prev = datetime.toDateTime();
         _for_each(providers, _tp, ITimeProvider *)
```
Comment (m-mora) [inline src/system/irrigation.h:85]: I think the priority was to use the NTP, if it fails then use RTC, if NTP works , then need to update RTC, not sure if this logic is doing that

File: src/system/irrigation.h (hunk 1 above)
Comment (Elemeants) [reply src/system/irrigation.h:97]: Yep, I think it should be easy.

File: src/system/irrigation.h (hunk 2 above)
Comment (Elemeants) [reply src/system/irrigation.h:85]: It takes the latest hour as the right one, and then updates the other providers with that hour.

**Pull request closed without merging**